test/integration/perfutil/util.go

```diff
@@ -0,0 +1,99 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package perfutil
+
+import (
+	"net/http"
+	"net/http/httptest"
+
+	"github.com/golang/glog"
+	"k8s.io/api/core/v1"
+	"k8s.io/client-go/informers"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/kubernetes/pkg/scheduler"
+	_ "k8s.io/kubernetes/pkg/scheduler/algorithmprovider"
+	"k8s.io/kubernetes/pkg/scheduler/factory"
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+	"k8s.io/kubernetes/test/integration/framework"
+)
+
+const enableEquivalenceCache = true
+
+type ShutdownFunc func()
+
+// StartMaster starts a local master for testing and returns the handle to the master URL
+// and the shutdown function to stop it.
+func StartTestMaster() (string, ShutdownFunc) {
+	h := &framework.MasterHolder{Initialized: make(chan struct{})}
+	s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		<-h.Initialized
+		h.M.GenericAPIServer.Handler.ServeHTTP(w, req)
+	}))
+
+	framework.RunAMasterUsingServer(framework.NewIntegrationTestMasterConfig(), s, h)
+
+	shutdownFunc := func() {
+		glog.Infof("destroying master")
+		s.Close()
+		glog.Infof("destroyed master")
+	}
+	return s.URL, shutdownFunc
+}
+
+// StartScheduler configures and starts a scheduler given a handle to the clientSet interface
+// and event broadcaster. It returns a handle to the configurator for the running scheduler
+// and the shutdown function to stop it.
+func StartScheduler(clientSet clientset.Interface, evtBroadcaster record.EventBroadcaster) (scheduler.Configurator, ShutdownFunc) {
+	informerFactory := informers.NewSharedInformerFactory(clientSet, 0)
+
+	schedulerConfigurator := factory.NewConfigFactory(
+		v1.DefaultSchedulerName,
+		clientSet,
+		informerFactory.Core().V1().Nodes(),
+		informerFactory.Core().V1().Pods(),
+		informerFactory.Core().V1().PersistentVolumes(),
+		informerFactory.Core().V1().PersistentVolumeClaims(),
+		informerFactory.Core().V1().ReplicationControllers(),
+		informerFactory.Extensions().V1beta1().ReplicaSets(),
+		informerFactory.Apps().V1beta1().StatefulSets(),
+		informerFactory.Core().V1().Services(),
+		informerFactory.Policy().V1beta1().PodDisruptionBudgets(),
+		informerFactory.Storage().V1().StorageClasses(),
+		v1.DefaultHardPodAffinitySymmetricWeight,
+		enableEquivalenceCache,
+	)
+
+	sched, err := scheduler.NewFromConfigurator(schedulerConfigurator, func(conf *scheduler.Config) {
+		conf.Recorder = evtBroadcaster.NewRecorder(legacyscheme.Scheme, v1.EventSource{Component: "scheduler"})
+	})
+	if err != nil {
+		glog.Fatalf("Error creating scheduler: %v", err)
+	}
+
+	stop := make(chan struct{})
+	informerFactory.Start(stop)
+
+	sched.Run()
+
+	destroyFunc := func() {
+		glog.Infof("destroying scheduler")
+		sched.StopEverything()
+		close(stop)
+		glog.Infof("destroyed scheduler")
+	}
+	return schedulerConfigurator, destroyFunc
+}
```

Comment:
I think you need to add a newline after this to get rid of that strange red symbol.